modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -664,48 +677,31 @@ private GridCoverage2D getMapReproject(
     CoordinateReferenceSystem getBestSourceCRS(
             CoordinateReferenceSystem targetCRS, WMTSReadParameters readParameters)
             throws IOException {
-        try {
-            CoordinateReferenceSystem result = null;
-            String switched = switchDefinition(targetCRS);
-            if (switched != null) {
-                // check first by changing the srs definition
-                result = CRS.decode(switched);
-                LOGGER.fine(() -> "Will request tiles with server supported crs " + switched);
-            } else if (readParameters.getSourceCRS() != null) {
-                // otherwise get the preferred one
-                result = readParameters.getSourceCRS();
-            } else if (!validSRS.isEmpty()) {
-                // last attempt is to retrieve the firs valid srs available
-                LOGGER.fine(() -> "Retrieving random srs to perform reprojection");
-                result = randomSupportedCRS();
```

Comment:
This is the part that should indeed go... as far as I can see, there is already a loop trying to decode all supported CRSs while initializing the srsName, doing it again here does not seem like it would help.
@taba90 can you confirm?

Reply:
@roarbra @aaime my apologies for the delay on this. 
Yes @aaime it is doing an unecessary loop over the srs as it has been already done before.